### PR TITLE
Reference Frame module abc2dq and dq2alphabeta

### PR DESCRIPTION
**Pull request by murgui**
It is missing to test with an actual sine as an input. I will do that after developing UART communication.

### CODE REVIEW

File: Software/Development/SemesterProject/App/Include/referenceFrameConversion.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+/*
+ * end of temporaty declarations
+ */
+
+
+typedef struct
+{
+    float DComponent;
+    float QComponent;
+}DQObject;
```
Comment (AitorTeran) [inline Software/Development/SemesterProject/App/Include/referenceFrameConversion.h]: dComponent, qComponent and dqObject?¿

Comment (AitorTeran) [reply Software/Development/SemesterProject/App/Include/referenceFrameConversion.h]: I see you called everything with capital letters, i think it's better with lower case but it's not a big deal so up to you

File: Software/Development/SemesterProject/App/referenceFrameConversion.c [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+ * This function performs the calculation of the alpha-beta components.
+ * It uses theta in radians and DQ components provided by controlTask.
+ * It returns a structure with both alpha and beta components.
+ *
+ * USE
+ *     alphaBetaObject abVoltage;
+ *     abVoltage = dq2alphabeta(&DQVoltage, theta);
+ *
+ */
+alphaBetaObject dq2alphabeta(DQObject *DQVoltage, float thetaRad)
```
Comment (AitorTeran) [inline Software/Development/SemesterProject/App/referenceFrameConversion.c]: Why DQVoltage? shouldn't it be more general in the functio, dqObject?

**Review by AitorTeran — APPROVED** [review]
Looking good, ready to be shipped!

File: Software/Development/SemesterProject/App/Include/referenceFrameConversion.h (hunk 1 above)
Comment (NicolaiFransen) [reply Software/Development/SemesterProject/App/Include/referenceFrameConversion.h]: I actually agree with Aitor¿?¿

File: Software/Development/SemesterProject/App/Include/referenceFrameConversion.h [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+typedef struct
+{
+    float alphaComponent;
+    float betaComponent;
+}alphaBetaObject;
```
Comment (NicolaiFransen) [inline Software/Development/SemesterProject/App/Include/referenceFrameConversion.h:21]: I think you should consider using afaBeta 🤔

**Review by NicolaiFransen — COMMENTED** [review]
Looks nice, but consider the capital letters

File: Software/Development/SemesterProject/App/Include/referenceFrameConversion.h (hunk 1 above)
Comment (murgui) [reply Software/Development/SemesterProject/App/Include/referenceFrameConversion.h]: Done

File: Software/Development/SemesterProject/App/referenceFrameConversion.c (hunk 2 above)
Comment (murgui) [reply Software/Development/SemesterProject/App/referenceFrameConversion.c]: Done

File: Software/Development/SemesterProject/App/Include/referenceFrameConversion.h (hunk 3 above)
Comment (murgui) [reply Software/Development/SemesterProject/App/Include/referenceFrameConversion.h:21]: ( ͡° ͜ʖ ͡°)

**Review by NicolaiFransen — APPROVED** [review]
Ship it (y)

**Pull request merged**